### PR TITLE
SearchBox: role=search is removed, but can be optionally defined via role prop.

### LINKED ISSUE #14914: SearchBox role is hardcoded and can not be changed.
### Environment Information

- **Package version(s)**: 7.128.1

### Describe the issue:

In some cases search box is a part of a larger region with a role 'search'. In this case role of the wrapper for a SearchBox need to be change to 'presentation'. However, setting role attribute on a SearchBox control has no effect, it remains "search" for a wrapper and "searchbox" for input.

Reason is that these roles are hard coded and can not be changed. This is the code from SearchBox.base.tsx

      <div **role="search**" ref={this._rootElement} className={classNames.root} onFocusCapture={this._onFocusCapture}>
        <div className={classNames.iconContainer} onClick={this._onClickFocus} aria-hidden={true}>
          <Icon iconName="Search" {...iconProps} className={classNames.icon} />
        </div>
        <input
          {...nativeProps}
          id={id}
          className={classNames.field}
          placeholder={placeholderValue}
          onChange={this._onInputChange}
          onInput={this._onInputChange}
          onBlur={this._onBlur}
          onKeyDown={this._onKeyDown}
          value={value}
          disabled={disabled}
          **role="searchbox"**
          aria-label={ariaLabel}
          ref={this._inputElement}
        />

At least it should be possible to change the role of the outer element.

#### Actual behavior:

Generated HTML for a SearchBox wrapper has a hard coded role of 'searchbox'.

#### Expected behavior:

Generated HTML for a SearchBox wrapper has a role attribute set as specified, e.g. "presentation".

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No)

Yes.

Requested priority: (Blocking, High, Normal, Low)

Normal.

Products/sites affected: (if applicable)

Word WAC.

### PR DESCRIPTION
**Pull request by dzearing**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #14914
- [x] Include a change request file using `$ yarn change`

## Before:

This JSX:
```
<SearchBox />
```
Renders this HTML: 
```jsx
<div role="search">
  <input ... />
</div>
```

## After:

This JSX:
```
<SearchBox />
<SearchBox role="search" />
```
Renders this HTML: 
```jsx
<div>
  <input ... />
</div>

<div role="search">
  <input ... />
</div>
```

### CODE REVIEW

**Comment by ecraig12345** [comment]
I think you have the wrong base branch.

**Comment by ecraig12345** [comment]
I mean it looks like your local branch for this PR was based off of master. It should be based off of 7.0.

**Comment by dzearing** [comment]
I think it's good now?

**Review by khmakoto — APPROVED** [review]
Seems like you are missing snapshot updates in `@fluentui/react-examples`.

**Review by Ridvanovskyy — APPROVED** [review]

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx [hunk 1]
```diff
@@ -76,6 +76,7 @@ export class SearchBoxBase extends React.Component<ISearchBoxProps, ISearchBoxSt
       clearButtonProps,
       disableAnimation,
       iconProps,
+      role,
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx:79]: Can you explicitly add this in ISearchBoxProps too?

**Review by Ridvanovskyy — APPROVED** [review]

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx (hunk 1 above)
Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx:79]: should be fixed now.

File: packages/react-next/src/components/SearchBox/SearchBox.types.ts [hunk 2]
```diff
@@ -108,6 +108,11 @@ export interface ISearchBoxProps
    */
   underlined?: boolean;
 
+  /**
+   * The role assigned to the root DIV element of the SearchBox, useful for defining a landmark role, such as "search".
+   */
+  role?: string;
```
Comment (ecraig12345) [inline packages/react-next/src/components/SearchBox/SearchBox.types.ts:114]: This should be in OUFR not react-next

Comment (dzearing) [reply packages/react-next/src/components/SearchBox/SearchBox.types.ts:114]: whoops, missed this. Fixing now!

**Pull request merged**